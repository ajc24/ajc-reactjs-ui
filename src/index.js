export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
