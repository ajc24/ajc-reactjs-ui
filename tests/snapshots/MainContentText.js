import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { MainContentText } from '../../src';

describe('MainContentText', () => {
  describe('Default props and rendering - Paragraph component', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentText>
            Test paragraph text.
          </MainContentText>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Heading component with upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentText type="heading" addUpperSpacing={true}>
            Test heading text
          </MainContentText>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Hyperlink component', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentText type="hyperlink" href="http://www.test.com">
            Test hyperlink
          </MainContentText>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Paragraph in italics component', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MainContentText type="paragraph-italics">
            Test paragraph in italics text.
          </MainContentText>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
