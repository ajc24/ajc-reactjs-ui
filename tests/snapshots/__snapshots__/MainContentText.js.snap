// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentText Default props and rendering - Paragraph component verifies the snapshot for the component 1`] = `
<p
  className="ajc-text-paragraph-container ajc-text-black ajc-default"
>
  Test paragraph text.
</p>
`;

exports[`MainContentText Transferred props and rendering - Heading component with upper spacing verifies the snapshot for the component 1`] = `
<p
  className="ajc-text-paragraph-container ajc-text-black ajc-default ajc-text-paragraph-upper-spacing"
>
  <b
    className="ajc-text-paragraph-heading"
  >
    Test heading text
  </b>
</p>
`;

exports[`MainContentText Transferred props and rendering - Hyperlink component verifies the snapshot for the component 1`] = `
<p
  className="ajc-text-paragraph-container ajc-text-black ajc-default"
>
  <a
    className="ajc-hyperlink"
    href="http://www.test.com"
  >
    Test hyperlink
  </a>
</p>
`;

exports[`MainContentText Transferred props and rendering - Paragraph in italics component verifies the snapshot for the component 1`] = `
<p
  className="ajc-text-paragraph-container ajc-text-black ajc-default"
>
  <i>
    Test paragraph in italics text.
  </i>
</p>
`;
